packages/classic/src/core/challenge.tsx

```diff
@@ -36,11 +37,13 @@ export class ChallengeService {
   #redis: RedisClient;
   #challengeToWordService: ChallengeToWordService;
   #challengeToPostService: ChallengeToPostService;
+  #mode: Mode;
 
-  constructor(redis: RedisClient) {
+  constructor(redis: RedisClient, mode: Mode) {
```

Comment:
suggestion: could default mode to regular mode by using a default value
```suggestion
  constructor(redis: RedisClient, mode: Mode = 'regular') {
```

Reply:
I think doing this is extremely error prone and I want to explain why:

Every time `ChallengeService` is used - it _needs_ to know whether it's in the context of `regular` or `hardcore`, and if that's **ever** determined incorrectly, it will result in bugs. Making this default to "regular" has the slight advantage of making it so this particular PR can be a little smaller - I won't need to update every instantiation. However - every instantiation does need to be updated before we ship - each and every one needs to do something to determine whether it's in the context of hardcore or in the context of regular - and changing them all now forces me to add TODO's for each of them to burn down in the future. I didn't want to do that in this PR because that's a harder thing to do (as each instantiation will need to be handled slightly differently) - but I really don't want mode to have a default long term so adding it in the immediate term doesn't seem like the right idea.